<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>simpleOscilloscope</class>
 <widget class="QMainWindow" name="simpleOscilloscope">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>804</width>
    <height>509</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>串口数据图表化</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res/images.qrc">
    <normaloff>:/general/general/graphsPolyline.png</normaloff>:/general/general/graphsPolyline.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="QGBSerialConfig">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>171</width>
        <height>211</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>171</width>
        <height>211</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox{
border-width:1px;
border-style:solid;
border-radius:4px;
}</string>
      </property>
      <property name="title">
       <string>串口设置</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <widget class="QPushButton" name="btnOpenPort">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>69</x>
         <y>165</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>打开串口</string>
       </property>
       <property name="icon">
        <iconset resource="../res/images.qrc">
         <normaloff>:/action/action/centrejust.png</normaloff>
         <normalon>:/action/action/process-stop.png</normalon>:/action/action/centrejust.png</iconset>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
       <property name="default">
        <bool>false</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QWidget" name="formLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>151</width>
         <height>136</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="commParam">
        <item row="3" column="0">
         <widget class="QLabel" name="lblVerify">
          <property name="text">
           <string>校验</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="cbbDataBit">
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="lblPortNum">
          <property name="text">
           <string>端口号</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="lblStopBit">
          <property name="text">
           <string>停止位</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="lblBaudRate">
          <property name="text">
           <string>波特率</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="cbbPortNum">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maxVisibleItems">
           <number>20</number>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
          <property name="minimumContentsLength">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="cbbVerify">
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="cbbStopBit">
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="cbbBaud">
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToContents</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="lblDataBit">
          <property name="text">
           <string>数据位</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QPushButton" name="btnRefresh">
       <property name="geometry">
        <rect>
         <x>11</x>
         <y>165</y>
         <width>51</width>
         <height>31</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>刷新端口号</string>
       </property>
       <property name="text">
        <string>刷新</string>
       </property>
       <property name="icon">
        <iconset resource="../res/images.qrc">
         <normaloff>:/action/action/reload.png</normaloff>:/action/action/reload.png</iconset>
       </property>
      </widget>
     </widget>
    </item>
    <item row="0" column="1" rowspan="4">
     <widget class="QCustomPlot" name="customPlot" native="true"/>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="maximumSize">
       <size>
        <width>171</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>曲线设置</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QRadioButton" name="rdbSinglePlot">
           <property name="text">
            <string>单一曲线</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="cbbDataType">
           <property name="toolTip">
            <string>单个数据点的数据类型</string>
           </property>
           <item>
            <property name="text">
             <string>uint8_t</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>uint16_t</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>uint32_t</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>int8_t</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>int16_t</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>int32_t</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>float</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QRadioButton" name="rdbMultiPlot">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>多条曲线</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spbNumberG">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>2</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QComboBox" name="btnCurveSel"/>
         </item>
         <item>
          <widget class="QToolButton" name="btnColorSel">
           <property name="toolTip">
            <string>设置颜色</string>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <widget class="QLineEdit" name="txtCurveRename">
         <property name="toolTip">
          <string>曲线名字</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>804</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSaveImages"/>
   <addaction name="actionClear"/>
   <addaction name="actionPaused"/>
   <addaction name="actionSelect"/>
   <addaction name="actionCoorTips"/>
   <addaction name="actionHelpFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSaveImages">
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/save.png</normaloff>:/general/general/save.png</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="toolTip">
    <string>保存图像</string>
   </property>
  </action>
  <action name="actionCharts">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/graphsPolyline.png</normaloff>
     <normalon>:/general/general/graphsHistogram.png</normalon>:/general/general/graphsPolyline.png</iconset>
   </property>
   <property name="text">
    <string>柱状图</string>
   </property>
   <property name="toolTip">
    <string>柱状图/折线图</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/Broom.png</normaloff>:/general/general/Broom.png</iconset>
   </property>
   <property name="text">
    <string>清空图表</string>
   </property>
  </action>
  <action name="actionPaused">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/pause.png</normaloff>
     <normalon>:/general/general/play.png</normalon>:/general/general/pause.png</iconset>
   </property>
   <property name="text">
    <string>暂停</string>
   </property>
   <property name="toolTip">
    <string>暂停绘图,但仍继续接收数据</string>
   </property>
  </action>
  <action name="actionSelect">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/selection.png</normaloff>
     <normalon>:/general/general/pointer_1.png</normalon>:/general/general/selection.png</iconset>
   </property>
   <property name="text">
    <string>选择放大</string>
   </property>
   <property name="toolTip">
    <string>放大选择区域</string>
   </property>
  </action>
  <action name="actionCoorTips">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/tips.png</normaloff>:/general/general/tips.png</iconset>
   </property>
   <property name="text">
    <string>坐标显示</string>
   </property>
  </action>
  <action name="actionHelpFile">
   <property name="icon">
    <iconset resource="../res/images.qrc">
     <normaloff>:/general/general/help.png</normaloff>:/general/general/help.png</iconset>
   </property>
   <property name="text">
    <string>使用说明</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>../../qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res/images.qrc"/>
 </resources>
 <connections/>
</ui>
